### PR TITLE
Rewrite

### PR DESCRIPTION
**Pull request by Lord-Valen**
compileCommandsFor initially didn't work in my use, so I went to fix it. Wound up doing this instead. 

Exposed `compileCommands` in `lib`, rather than as a package. Rewrote `compileCommands` to accomodate use as a function. Used the Standard framework for simplicity. Formatted files.

### CODE REVIEW

File: comb/repo/devshells.nix [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+}: let
+  inherit (inputs) std;
+  lib = inputs.nixpkgs.lib // builtins;
+in
+  lib.mapAttrs (_: std.lib.dev.mkShell) {
+    default = {
+      name = "CreateCommands";
+      nixago = with cell.configs;
+        [
+          treefmt
```
Comment (kngrektor) [inline comb/repo/devshells.nix:13]: I think a formatter is a good idea

File: comb/repo/devshells.nix [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+        ]
+        ++ (with inputs.std-data-collection.data.configs; [
+          editorconfig
+          conform
+          lefthook
```
Comment (kngrektor) [inline comb/repo/devshells.nix:18]: I don't know what conform and lefthook is but it doesn't seem relevant to this repo, at least not at the moment.

I'm normally a large proponent of editorconfig but: I'd prefer a handwritten one that only cares about .nix and maybe .md, and it might be redundant if there's a formatter.

If we get rid of editorconfig, conform and lefthook here it means one less flake input which to me is a win.

File: flake.nix [hunk 3]
```diff
@@ -1,12 +1,35 @@
 {
   inputs = {
     nixpkgs.url = "github:NixOS/nixpkgs/nixos-unstable";
-    flake-utils.url = "github:numtide/flake-utils";
+    std = {
+      url = "github:divnix/std";
```
Comment (kngrektor) [inline flake.nix:5]: This is the first time I've really looked at std this close so I'm not really comfortable but it doesn't look too bad

File: comb/compileCommands/lib.nix [hunk 4]
```diff
@@ -0,0 +1,22 @@
+{
+  inputs,
+  cell
+}: {
+  compileCommands = pkg:
+    pkg.overrideAttrs (super: {
+      installPhase = "cp compile_commands.json $out";
+      cmakeFlags = super.cmakeFlags ++ ["-DCMAKE_EXPORT_COMPILE_COMMANDS=1"];
```
Comment (kngrektor) [inline comb/compileCommands/lib.nix:8]: I wonder how CMAKE_EXPORT_COMPILE_COMMANDS works in combination with cc-wrapper-hook.

I assume worst case scenario is duplicated commands in compile_commands.json but I hope it never runs cc-wrapper-hook.

**Review by kngrektor — COMMENTED** [review]
First time doing a code review, hope it's not too bad.

I haven't had time to test it but will hopefully get around to it today.

File: comb/repo/devshells.nix (hunk 2 above)
Comment (Lord-Valen) [reply comb/repo/devshells.nix:18]: I should make it clear that `std-data-collection` only provides nixago presets. Conform enforces commit message rules; the [preset](https://github.com/divnix/std-data-collection/blob/main/data/configs/conform.nix) uses conventional commits. Lefthook is a pre-commit task-runner; the [preset](https://github.com/divnix/std-data-collection/blob/main/data/configs/lefthook.nix) runs treefmt and conform. If you want to manually write the editorconfig, it can be done in configs.nix (though what you're after is expressed in the [preset](https://github.com/divnix/std-data-collection/blob/main/data/configs/editorconfig.nix) anyway).

If you want to get rid of conform, I would recommend keeping lefthook (but not with the preset).

File: comb/compileCommands/lib.nix (hunk 4 above)
Comment (Lord-Valen) [reply comb/compileCommands/lib.nix:8]: > I assume worst case scenario is duplicated commands in compile_commands.json but I hope it never runs cc-wrapper-hook.

That does seem to be the case.

File: comb/repo/devshells.nix (hunk 2 above)
Comment (kngrektor) [reply comb/repo/devshells.nix:18]: Seems I have done the classic mixing-up-git-and-github error, I read github hook and thought it was related to github actions somehow.
This makes a lot more sense and looks good :+1:
